modules/core/shared/src/main/scala/gem/util/Timestamp.scala

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.util
+
+import cats._
+import cats.implicits._
+import cats.effect.IO
+
+import java.time.{ Instant, ZonedDateTime }
+import java.time.ZoneOffset.UTC
+import java.time.temporal.ChronoUnit.MICROS
+
+
+/** Timestamp wraps a `java.util.Instant` that is truncated to microsecond
+  * resolution.  This allows Timestamps to roundtrip to/from the database
+  * where only microsecond resolution is supported.  In addition the min
+  * and max instants that are supported are determined by the postgres limit for
+  * timestamps.
+  *
+  * @param toInstant
+  */
+final class Timestamp private(val toInstant: Instant) extends AnyVal {
+
```

Comment:
Can we add range and scale assertions here as a sanity check, as we did for a few other types?

Reply:
Apparently assertions aren't legal in value classes, but I could just lose the value class-ness of it.

Reply:
Oh I didn't notice it was a value class. Yeah I would be ok dropping the `extends AnyVal`.

Reply:
I just went with the `sealed abstract case class` trick to avoid having to write `equals` and `hashCode`, which would have required an `Eq` instance for `Instant` to be defined somewhere due to the `[wartremover:Equals]` check.